### PR TITLE
[Docu] DevTutorial: Text Proposal Persistence

### PR DESCRIPTION
**Pull request by fneumann**
Proposal for the tutorial on persistence in ILIAS. Reworked and translated the initial German text to English.

### CODE REVIEW

**Comment by acgruber** [comment]
Hi @chfsx,
would you be so kind to review this sometime soon so I can merge? :)
Thanks a lot!
Ann-Christin

File: docs/development/devguide/tutorial/04-coding/08-persistence-data.md
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/08-persistence-data.md]:
```suggestion
## Why do I need This in ILIAS?
```

File: docs/development/devguide/tutorial/04-coding/08-persistence-data.md
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/08-persistence-data.md]:
```suggestion
## How to proceed?
```

File: docs/development/devguide/tutorial/04-coding/08-persistence-data.md
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/08-persistence-data.md]:
```suggestion
## What is this about?
```

File: docs/development/devguide/tutorial/04-coding/08-persistence-data.md
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/08-persistence-data.md]:
```suggestion
# Database usage in ILIAS
```

File: docs/development/devguide/tutorial/04-coding/08-persistence-data.md
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/08-persistence-data.md]:
```suggestion
ILIAS is designed for the use of MySQL or MariaDB as database systems, but also supported Oracle and PostgreSQL in former versions. To keep this flexibility, database calls must use the interface [ilDBInterface](../../../../../components/ILIAS/Database/interfaces/interface.ilDBInterface.php), whose functions largely abstract the peculiarities of certain SQL dialects.
```

File: docs/development/devguide/tutorial/04-coding/08-persistence-data.md
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/08-persistence-data.md]:
```suggestion
To access the ILIAS database, the classes of your component also requires an instance of the global database class. If it cannot be passed as a parameter of the constructor, it should be taken from the dependency injection container of ILIAS, which is available as a global variable `$DIC`.
```

File: docs/development/devguide/tutorial/04-coding/08-persistence-data.md
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/08-persistence-data.md]: maybe you could explain that `nextId` is only possible if the tables has been created with `createSequence`

File: docs/development/devguide/tutorial/04-coding/08-persistence-data.md
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/08-persistence-data.md]:
```suggestion
## What do I need to watch out for? (Dos & Dont's)
```

File: docs/development/devguide/tutorial/04-coding/TodoExample/classes/class.ilTodoItem.php
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/TodoExample/classes/class.ilTodoItem.php]: Those lines can be removed, see below

File: docs/development/devguide/tutorial/04-coding/TodoExample/classes/class.ilTodoItem.php
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/TodoExample/classes/class.ilTodoItem.php]:
```suggestion
        private int $todo_id,
        private int $user_id,
        private string $title,
        private ?string $description,
        private ?string $deadline
```

File: docs/development/devguide/tutorial/04-coding/TodoExample/classes/class.ilTodoItem.php
Comment (chfsx) [inline docs/development/devguide/tutorial/04-coding/TodoExample/classes/class.ilTodoItem.php]: this lines can be removed with the suggestion above

**Review by chfsx — CHANGES_REQUESTED** [review]
@fneumann Thanks a lot for the proposal, I made several inline comments and suggestion. 

One general request:

Please use namespaces for the example files and rename the `classes` to `src`. I think we are more state of the art then since we are committing this tutorial for trunk with the ongoing component revision.

namespace would be `ILIAS\components\Todo then.

@acgruber I wait for the feedback of @fneumann and the changes requested, after that I'll update my review and will give you my go for the merge. thanks in advance!

**Comment by acgruber** [comment]
Thanks a lot for your review and suggestions, @chfsx!

@fneumann, would be great if you found the time to look at this sometime soon :)

**Comment by fneumann** [comment]
Sorry for the delay. No I worked in all suggestions.

**Comment by acgruber** [comment]
Thank you, @fneumann!

@chfsx, I will merge as soon as you approve the changes!

**Comment by acgruber** [comment]
Hello @fneumann and @chfsx,
Could you kindly give me an update on this PR? Can this be merged?
Thanks!

**Comment by fneumann** [comment]
Missed one little wording issue. Now everything should be resolved. @chfsx would you approve this?

**Comment by chfsx** [comment]
Many thanks @fneumann  for your great work! every documentation brings ILIAS forward and is very valuable. thank you also for your patience in this matter, I am sorry that the PR remained open for so long.

**Pull request merged**
